### PR TITLE
Add optional headers to utm modules

### PR DESCRIPTION
**Pull request by MatrixCrawler**
##### SUMMARY
This PR will add optional header to the utm modules POST and PUT requests as it is needed for some modules.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- module_utils/utm_utils

### CODE REVIEW

File: lib/ansible/utils/module_docs_fragments/utm.py [hunk 1]
```diff
@@ -35,10 +40,8 @@ class ModuleDocFragment(object):
           - The desired state of the object.
           - C(present) will create or update an object
           - C(absent) will delete an object if it was present
-          - C(info) will return the object details
```
Comment (Akasurde) [inline lib/ansible/utils/module_docs_fragments/utm.py:38]: If you are going to remove this then you might want to remove [this](https://github.com/ansible/ansible/blob/59dcf3f27743ecbecfadc037515d0fa33ae3045a/lib/ansible/module_utils/utm_utils.py#L67) as well.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please add unit test for this change - [here](https://github.com/ansible/ansible/tree/devel/test/units/modules/web_infrastructure) ? Thanks.

File: test/units/module_utils/test_utm_utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
```
Comment (Akasurde) [inline test/units/module_utils/test_utm_utils.py]: Sorry for not mentioning this before, could you please write this test case using `pytest` ?

Comment (MatrixCrawler) [reply test/units/module_utils/test_utm_utils.py]: Of course.

JFYI: The documentation show a wrong or outdated example then (with unittest)
https://docs.ansible.com/ansible/latest/dev_guide/testing_units_modules.html#a-complete-example

**Review by maunzCache — APPROVED** [review]

File: test/units/module_utils/test_utm_utils.py (hunk 2 above)
Comment (maunzCache) [reply test/units/module_utils/test_utm_utils.py]: @Akasurde Fine with you now? Can we shipit ?

**Review by Akasurde — APPROVED** [review]

**Comment by MatrixCrawler** [comment]
shipit

**Comment by Akasurde** [comment]
@MatrixCrawler Thanks for the contribution. @getjack Thanks for the review.

**Pull request merged**